client/src/main/scala/org/http4s/client/JavaNetClient.scala

```diff
@@ -0,0 +1,186 @@
+  * exploration in a REPL.
+  *
+  * All I/O operations in this client are blocking.
+  *
+  * @param blockingExecutionContext An `ExecutionContext` on which
+  * blocking operations will be performed.
+  * @param ec The `ExecutionContext` to which work will be shifted
+  * back after blocking.
+  */
+sealed abstract class JavaNetClient private (
+    val connectTimeout: Duration,
+    val readTimeout: Duration,
+    val proxy: Option[Proxy],
+    val hostnameVerifier: Option[HostnameVerifier],
+    val sslSocketFactory: Option[SSLSocketFactory],
+    val blockingExecutionContext: ExecutionContext
+) {
+  private def copy(
+      connectTimeout: Duration = connectTimeout,
+      readTimeout: Duration = readTimeout,
+      proxy: Option[Proxy] = proxy,
+      hostnameVerifier: Option[HostnameVerifier] = hostnameVerifier,
+      sslSocketFactory: Option[SSLSocketFactory] = sslSocketFactory,
+      blockingExecutionContext: ExecutionContext = blockingExecutionContext
+  ): JavaNetClient =
+    new JavaNetClient(
+      connectTimeout = connectTimeout,
+      readTimeout = readTimeout,
+      proxy = proxy,
+      hostnameVerifier = hostnameVerifier,
+      sslSocketFactory = sslSocketFactory,
+      blockingExecutionContext = blockingExecutionContext
+    ) {}
+
+  def withConnectTimeout(connectTimeout: Duration): JavaNetClient =
+    copy(connectTimeout = connectTimeout)
+
+  def withReadTimeout(readTimeout: Duration): JavaNetClient =
+    copy(readTimeout = readTimeout)
+
+  def withProxyOption(proxy: Option[Proxy]): JavaNetClient =
+    copy(proxy = proxy)
+  def withProxy(proxy: Proxy): JavaNetClient =
+    withProxyOption(Some(proxy))
+  def withoutProxy: JavaNetClient =
+    withProxyOption(None)
+
+  def withHostnameVerifierOption(hostnameVerifier: Option[HostnameVerifier]): JavaNetClient =
+    copy(hostnameVerifier = hostnameVerifier)
+  def withHostnameVerifier(hostnameVerifier: HostnameVerifier): JavaNetClient =
+    withHostnameVerifierOption(Some(hostnameVerifier))
+  def withoutHostnameVerifier: JavaNetClient =
+    withHostnameVerifierOption(None)
+
+  def withSslSocketFactoryOption(sslSocketFactory: Option[SSLSocketFactory]): JavaNetClient =
+    copy(sslSocketFactory = sslSocketFactory)
+  def withSslSocketFactory(sslSocketFactory: SSLSocketFactory): JavaNetClient =
+    withSslSocketFactoryOption(Some(sslSocketFactory))
+  def withoutSslSocketFactory: JavaNetClient =
+    withSslSocketFactoryOption(None)
+
+  def withBlockingExecutionContext(blockingExecutionContext: ExecutionContext): JavaNetClient =
+    copy(blockingExecutionContext = blockingExecutionContext)
+
+  /** Creates the `JavaNetClient`.
+    *
+    * The shutdown of this client is a no-op.  Creation of the client
+    * allocates no resources, and any resources allocated while using
+    * this client are reclaimed by the JVM at its own leisure.
+    */
+  def create[F[_]](implicit F: Async[F], timer: Timer[F]): Client[F] = Client(open, F.unit)
```

Comment:
Not sure if taking the timer here is the right thing.  In 0.18, the EC is taken on the class itself, but that would require making the whole builder polymorphic.

Reply:
`Timer` comes from `cats 1.x` right?